docs/Release_Management.md

```diff
@@ -0,0 +1,50 @@
+# Release Management
+
+## Overview
+This document describes **Kubernetes Secrets Store CSI Driver** project release management, which talks about versioning, branching and cadence.
+
+## Legend
+
+- **X.Y.Z** refers to the version (git tag) of Secrets Store CSI Driver that is released. This is the version of the Secrets Store CSI Driver image.
+ 
+- **Milestone** should be designed to include feature sets to accommodate monthly release cycles including test gates. GitHub milestones are used by maintainers to manage each release. PRs and Issues for each release should be created as part of a corresponding milestone.
+
+- **Test gates** should include soak tests and upgrade tests from the last minor version.
+
+
+## Versioning
+This project strictly follows [semantic versioning](https://semver.org/spec/v2.0.0.html). All releases will be of the form _vX.Y.Z_ where X is the major version, Y is the minor version and Z is the patch version.
+
+### Patch releases
+- Patch releases provide users with bug fixes and security fixes. They do not contain new features.
+
+### Minor releases
+- Minor releases contain security and bug fixes as well as _**new features**_.
+
+- They are backwards compatible.
```

Comment:
Are we committing to backwards compatible within the Major release? 

Does backwards compatible imply upgrade path or just features?

I'm not sure that we would want to commit to backwards compatibility until v1 Major release and within a Major release I'm not sure we want to support arbitrary upgrades.

example:

* Should `0.2.0` be backwards compatible in features with `0.1.0`
* Should a direct upgrade from `1.1.0` to `1.3.0` be possible or may you need to go through `1.1.0 -> 1.2.0 -> 1.3.0` (from the test gate section above i assume that the user would need to step through each minor release)

Reply:
I think major release should imply breaking change. For eg if we go from `1.0.0` -> `2.0.0` then we indicate there is breaking change and not all features would work as expected.

Also, backward compatibility I am thinking in terms of features and not upgrade path. 
>Should 0.2.0 be backwards compatible in features with 0.1.0

This is correct

>Should a direct upgrade from 1.1.0 to 1.3.0 be possible or may you need to go through 1.1.0 -> 1.2.0 -> 1.3.0 (from the test gate section above i assume that the user would need to step through each minor release)

I think it should be possible to go from `1.1.0` -> `1.3.0`. I don't believe we have any such restriction today. (Please correct me if I am wrong here)

As for backward compatibility we can document since which version/release we can support. WDYT?

Reply:
ACK, I'm not sure its feasible for us to maintain and test all permutations of up update paths, so maybe just adding some wording that upgrades are only _tested_ for consecutive version numbers

Reply:
@tam7t  / @aramase Do we do any upgrade test for driver?

Reply:
Since you're working on #457 - yes we intend to so this sgtm